### ISSUE TITLE
Suggested fix columns in JSON output may be off by 1

### ISSUE THREAD

**Issue by tasgon** [charliermarsh/ruff#3106]
Consider the following file:
```py
import sys, json # comment
sys.exit(42)
```

Running `ruff check --output json --select F401` on that yields:
```
[
  {
    "code": "F401",
    "message": "`json` imported but unused",
    "fix": {
      "content": "import sys",
      "message": "Remove unused import: `json`",
      "location": {
        "row": 1,
        "column": 0
      },
      "end_location": {
        "row": 1,
        "column": 16
      }
    },
    "location": {
      "row": 1,
      "column": 13
    },
    "end_location": {
      "row": 1,
      "column": 17
    },
    "filename": "test.py"
  }
]
```

Is this intended, or should the column fields in the `fix` object be 1 and 17, respectively?

**Comment by charliermarsh** [comment]
Yeah this is bad but "known" (and existing clients, like the LSP, rely on it). It happened unintentionally. RustPython returns locations with a one-indexed row and a zero-indexed column. When we create those top-level locations, we modify the column to be one-indexed, which is more natural when reporting. However, we forgot to do the same for the fix locations (which don't appear in the standard output, so it was an oversight).

**Comment by charliermarsh** [comment]
You're welcome to rely on these semantics as other clients already do so. Perhaps I'll fix this as a breaking change when we go to 1.0.

**Comment by youknowone** [comment]
Is this a bug of Location or only related to spec of json output?

**Comment by charliermarsh** [comment]
@youknowone - It's a bug on our end -- we're just not consistent about using zero- or one-based indexing for these, but @MichaReiser has a fix for it on our side.

**Comment by charliermarsh** [comment]
I believe this was fixed by #4007.
